File: sort/bubble_sort/Java/BubbleSort.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        int arr[] = {10, 8, 7, 6, 5, 2, 2};
        bubbleSort(arr);
        System.out.println(Arrays.toString(arr));


    }

    public static void bubbleSort(int[] numArray) {

        int n = numArray.length;
        int temp = 0;

        for (int i = 0; i < n; i++) {
            for (int j = 1; j < (n - i); j++) {

                if (numArray[j - 1] > numArray[j]) {
                    temp = numArray[j - 1];
                    numArray[j - 1] = numArray[j];
                    numArray[j] = temp;
                }

            }
        }
    }
}

File: sort/bubble_sort/C++/bubble_sort.cpp
#include<iostream>
using namespace std;
int main()
{
	long long n,a[100000],temp;
	cin>>n;
	for(int i=0;i<n;i++)
	  cin>>a[i];
	for(int i=n-1;i>0;i--)
	{
		 for(int j=0;j<i;j++)
		 if(a[j]>a[j+1])
		 {
			 temp=a[j];
			 a[j]=a[j+1];
			 a[j+1]=temp;
		 }
	}
	for(int i=0;i<n;i++)
	   cout<<a[i]<<' ';
return 0;
}